### PR TITLE
Support baubles with shields and update to ObfuscationReflectionHelper

### PR DESCRIPTION
**Pull request by FoxMcloud5655**
Update to support baubles implementing the ICustomArmor class to provide shields.

### CODE REVIEW

File: src/main/java/com/brandon3055/draconicevolution/items/Debugger.java [hunk 1]
```diff
@@ -1,14 +1,19 @@
 package com.brandon3055.draconicevolution.items;
```
Comment (brandon3055) [inline src/main/java/com/brandon3055/draconicevolution/items/Debugger.java]: Changes to this file are unnecessary

File: src/main/java/com/brandon3055/draconicevolution/handlers/CustomArmorHandler.java [hunk 2]
```diff
@@ -130,14 +135,31 @@ public void onPlayerAttacked(LivingAttackEvent event) {
...
+        replaceBaubles(summery, player);
         if (remainingPoints > 0) {
             player.hurtResistantTime = 20;
         }
         else if (hitAmount - totalAbsorbed > 0) {
             player.attackEntityFrom(event.getSource(), hitAmount - totalAbsorbed);
         }
     }
+    
+    private void replaceBaubles(ArmorSummery summery, EntityPlayer player) {
```
[the remark this answers is not in the document]
Comment (brandon3055) [reply src/main/java/com/brandon3055/draconicevolution/handlers/CustomArmorHandler.java]: Oh one more thing. Your code will crash if baubles is not installed.

**Comment by brandon3055** [comment]
Ok so i went ahead and added baubles support myself. And while i was at it i went ahead and swapped out ReflectionHelper
https://github.com/brandon3055/Draconic-Evolution/commit/f58d6fef804b6d3d75252560b1270e68aea63046

File: src/main/java/com/brandon3055/draconicevolution/handlers/CustomArmorHandler.java (hunk 2 above)
[the remark this answers is not in the document]
Comment (FoxMcloud5655) [reply src/main/java/com/brandon3055/draconicevolution/handlers/CustomArmorHandler.java]: I see.  Sorry, I just got into modding as of a month ago, so I'm not privy to the nuances of how to handle things properly.

File: src/main/java/com/brandon3055/draconicevolution/client/gui/modwiki/moddata/displaycontent/DCImage.java [hunk 3]
```diff
@@ -131,7 +134,7 @@ else if (resourceLocation.dlFailed) {
             }
             else if (resourceLocation.dlFinished && mouseButton == 1) {
                 try {
-                    ReflectionHelper.setPrivateValue(GuiScreen.class, branch.guiWiki, new URI(url), "clickedLinkURI", "field_175286_t");
+                    ObfuscationReflectionHelper.setPrivateValue(GuiScreen.class, branch.guiWiki, new URI(url), "clickedLinkURI");
```
[the remark this answers is not in the document]
Comment (FoxMcloud5655) [reply src/main/java/com/brandon3055/draconicevolution/client/gui/modwiki/moddata/displaycontent/DCImage.java]: Okay, I can see that.  I just saw that it was depreciated and that one of the issues had to deal with crashes related to using ReflectionHelper.

File: src/main/java/com/brandon3055/draconicevolution/handlers/CustomArmorHandler.java [hunk 4]
```diff
@@ -449,17 +473,29 @@ private static boolean applyArmorDamageBlocking(LivingAttackEvent event, ArmorSu
         public boolean hasDraconic = false;
 
         public ArmorSummery getSummery(EntityPlayer player) {
+            List<ItemStack> slots = new ArrayList<ItemStack>();
             NonNullList<ItemStack> armorSlots = player.inventory.armorInventory;
+            for (int i = 0; i < armorSlots.size(); i++) {
+                slots.add(armorSlots.get(i));
+            }
+            if (Loader.isModLoaded("baubles")) {
```
[the remark this answers is not in the document]
Comment (FoxMcloud5655) [reply src/main/java/com/brandon3055/draconicevolution/handlers/CustomArmorHandler.java]: Ah, I was wondering why ModHelper had those stored; didn't know it was for performance reasons.

File: src/main/java/com/brandon3055/draconicevolution/items/Debugger.java (hunk 1 above)
Comment (FoxMcloud5655) [reply src/main/java/com/brandon3055/draconicevolution/items/Debugger.java]: Whoops.  I think that was my fault; I meant to have that in another fork.

**Comment by FoxMcloud5655** [comment]
Hey, I'm totally fine with that!  Thanks for all of your hard work.  I'll try not to be as obsessive :P

**Pull request closed without merging**